transformer_engine/common/include/transformer_engine/activation.h

```diff
@@ -17,6 +17,23 @@
 extern "C" {
 #endif
 
+/*! \enum NVTE_Activation_Type
+ *  \brief TE activation function type.
+ */
+enum class NVTE_Activation_Type {
+  GELU      = 0,  /*!< Gaussian Error Linear Unit */
+  GEGLU     = 1,  /*!< Gated Gaussian Error Linear Unit */
+  SILU      = 2,  /*!< Sigmoid Linear Unit */
+  SWIGLU    = 3,  /*!< Gated Sigmoid Linear Unit */
+  RELU      = 4,  /*!< Rectified Linear Unit */
+  REGLU     = 5,  /*!< Gated Rectified Linear Unit */
+  QGELU     = 6,  /*!< Quick-GeLU Approximation */
+  QGEGLU    = 7,  /*!< Gated Quick-GeLU Approximation */
+  SRELU     = 8,  /*!< S-Shaped ReLU */
+  SREGLU    = 9,  /*!< Gated S-Shaped ReLU */
+};
```

Comment:
Enum classes are C++:
```suggestion
#ifdef __cplusplus
/*! \enum NVTE_Activation_Type
 *  \brief TE activation function type.
 */
enum class NVTE_Activation_Type {
  GELU      = 0,  /*!< Gaussian Error Linear Unit */
  GEGLU     = 1,  /*!< Gated Gaussian Error Linear Unit */
  SILU      = 2,  /*!< Sigmoid Linear Unit */
  SWIGLU    = 3,  /*!< Gated Sigmoid Linear Unit */
  RELU      = 4,  /*!< Rectified Linear Unit */
  REGLU     = 5,  /*!< Gated Rectified Linear Unit */
  QGELU     = 6,  /*!< Quick-GeLU Approximation */
  QGEGLU    = 7,  /*!< Gated Quick-GeLU Approximation */
  SRELU     = 8,  /*!< S-Shaped ReLU */
  SREGLU    = 9,  /*!< Gated S-Shaped ReLU */
};
#endif
```

Reply:
Do we or anyone else ever compile the core TE library as pure C?

If there's a need to support that, we should probably convert these enum classes to typedef'd enums. PyBind doesn't have any issues with that.

Reply:
In practice I don't think so, but once we add C++ it'll be hard to remove. Staying in pure C also helps avoid C++ ABI issues.